src/Microsoft.ML.Transforms/MissingValueDroppingTransformer.cs

```diff
@@ -65,275 +74,321 @@ private static VersionInfo GetVersionInfo()
-                    indices = new int[newCount];
+                Host.AssertValue(isNA);
 
-                int offset = 0;
+                int newCount = 0;
                 for (int i = 0; i < src.Count; i++)
                 {
                     if (!isNA(in src.Values[i]))
+                        newCount++;
+                }
+                Host.Assert(newCount <= src.Count);
+
+                if (newCount == 0)
+                {
+                    dst = new VBuffer<TDst>(src.Length - src.Count, 0, dst.Values, dst.Indices);
+                    return;
+                }
+
+                if (newCount == src.Count)
+                {
+                    Utils.Swap(ref src, ref dst);
+                    return;
+                }
+
+                var values = dst.Values;
+                if (Utils.Size(values) < newCount)
+                    values = new TDst[newCount];
+
+                int iDst = 0;
+                if (src.IsDense)
+                {
+                    for (int i = 0; i < src.Count; i++)
                     {
-                        values[iDst] = src.Values[i];
-                        indices[iDst] = src.Indices[i] - offset;
-                        iDst++;
+                        if (!isNA(in src.Values[i]))
+                        {
+                            values[iDst] = src.Values[i];
+                            iDst++;
+                        }
                     }
-                    else
-                        offset++;
+                    Host.Assert(iDst == newCount);
+                    dst = new VBuffer<TDst>(newCount, values, dst.Indices);
                 }
-                Host.Assert(iDst == newCount);
-                Host.Assert(offset == src.Count - newCount);
-                dst = new VBuffer<TDst>(src.Length - offset, newCount, values, indices);
+                else
+                {
+                    var indices = dst.Indices;
+                    if (Utils.Size(indices) < newCount)
+                        indices = new int[newCount];
+
+                    int offset = 0;
+                    for (int i = 0; i < src.Count; i++)
+                    {
+                        if (!isNA(in src.Values[i]))
+                        {
+                            values[iDst] = src.Values[i];
+                            indices[iDst] = src.Indices[i] - offset;
+                            iDst++;
+                        }
+                        else
+                            offset++;
+                    }
+                    Host.Assert(iDst == newCount);
+                    Host.Assert(offset == src.Count - newCount);
+                    dst = new VBuffer<TDst>(src.Length - offset, newCount, values, indices);
+                }
+            }
+        }
+    }
+    /// <summary>
+    /// Drops missing values from columns.
+    /// </summary>
+    public sealed class MissingValueDroppingEstimator : TrivialEstimator<MissingValueDroppingTransformer>
```

Comment:
>MissingValueDroppingEstimator [](start = 24, length = 29)

Add it to the Xtensions catalog, and if you are feeling generous write a sample for it.

Reply:
@Zruty0  I don't think we ever wanted to expose it, and I'm mainly doing it to properly convert WordBag transform into estimator chain.

Reply:
Yea, it's a wacky one, we can live without an `MLContext` extension for it

---
In reply to: [233947755](https://github.com/dotnet/machinelearning/pull/1624#discussion_r233947755) [](ancestors = 233947755)

Reply:
Or just add it to the extensions catalog  for this PR. 

---
In reply to: [233719342](https://github.com/dotnet/machinelearning/pull/1624#discussion_r233719342) [](ancestors = 233719342)